Content.Client/_RMC14/Xenonids/UI/XenoChoiceControl.xaml.cs

```diff
@@ -26,4 +43,68 @@ public void Set(FormattedMessage msg, Texture? texture)
         NameLabel.SetMessage(msg);
         Texture.Texture = texture;
     }
+
+    public void SetName(string name)
+    {
+        Button.Name = name.Replace(" ","");
+    }
+
+    public void SetLeader(Texture? texture)
+    {
+        LeaderStar.Texture = texture;
+        LeaderStar.Visible = true;
+        LeaderBox.Visible = true;
+        LeaderBar.Visible = true;
+    }
+
+    public void SetHealth(float health)
+    {
+        Health.Value = health;
+        if (health <= 0)
+            Health.BackgroundStyleBoxOverride = new StyleBoxFlat(Color.Red);
+        else
+            Health.BackgroundStyleBoxOverride = new StyleBoxFlat(Color.Gray);
+
+        Health.Visible = true;
+    }
+
+    public void SetPlasma(float plasma)
+    {
+        Plasma.Value = plasma;
+        Plasma.Visible = true;
+    }
+
+    public void SetEvo(int evo)
+    {
+        EvoPoints.Text = $"Evo: {evo}";
+        EvoPoints.Visible = true;
+    }
+
+
+    public void OnHovered(GUIMouseHoverEventArgs args)
+    {
+        if (args.SourceControl is not Button button)
+            return;
+
+            button.ModulateSelfOverride = (button.Name) switch
+            {
+                ("HealButton") => HealHighlightColor,
+                ("PlasmaButton") => PlasmaHighlightColor,
```

Comment:
You should be comparing this against the value of the buttons, make them access public if you have to, not their name as a string, and do not throw if its none of them, just dont set the button color